core/src/io/anuke/mindustry/world/blocks/power/PowerGraph.java

```diff
@@ -26,6 +28,8 @@
     private final int graphID;
     private static int lastGraphID;
 
+    private Color led = null;
```

Comment:
Instead of just having a single color, how about:
- an enum, `PowerState`, which would be `insufficient` / `producing` / `consuming` / `none` (or something similar). Each enum would have a `Color` assigned in its constructor. The empty state would have `Color.clear` (**not** null).
- assign the state each frame instead of the color.
- keep the color, but name it `statusColor` or something and `statusColor.lerp(status.color, 0.1f * Time.delta())` every frame so it changes to the current status color.
- add a getter for the status for possible future applications